src/components/Domain/addDomain.tsx

```diff
@@ -15,31 +19,50 @@ import {
   DialogDescription,
 } from "@/components/ui/dialog";
 import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+import { Textarea } from "@/components/ui/textarea";
 import {
   Select,
   SelectTrigger,
   SelectItem,
   SelectValue,
   SelectContent,
 } from "@/components/ui/select";
-import { Textarea } from "@/components/ui/textarea";
 
 interface DomainData {
-  domain: string;
+  _id: string;
+  label: string;
   description: string;
+  createdAt?: string; // Optional fields
+  createdBy?: string;
+  status?: string;
 }
+
 interface Domain {
+  _id: string;
   label: string;
+  description: string;
 }
 
+interface AddDomainProps {
+  onAddDomain: (newDomain: DomainData) => void; // Prop to pass the new domain
+}
+
+// Zod schema for form validation
 const domainSchema = z.object({
-  domain: z.string().nonempty("Please select a domain"),
+  label: z.string().nonempty("Please enter a domain name"),
   description: z.string().nonempty("Please enter a description"),
+  status: z.enum(["Active"]).default("Active"),
 });
 
-const AddDomain: React.FC = () => {
+const AddDomain: React.FC<AddDomainProps> = ({ onAddDomain }) => {
   const [open, setOpen] = useState(false);
-  const [domains, setDomains] = useState<Domain[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [successMessage, setSuccessMessage] = useState<string | null>(null);
+  const [domains, setDomains] = useState<Domain[]>([]); // Use Domain type here
+  const currentUserId = "user-id-123";
```

Comment:
why is this hardcoded here

Reply:
there is no navigation till now the generic code is just below it

Reply:
Is there no record id recorded for current logged in user?